demo/src/store/store.js

```diff
@@ -22,17 +22,17 @@ function reducer(state = {}, action) {
     case 'DECREMENT':
       item.quantity--;
       if (item.quantity < 0) {
-        console.error('Setting quantity to a negative amount!');
-        item.quantity++;
+        // Set quantity to 0 as negatives make no sense.
```

Comment:
A console.warn might still be appropriate here, since this indicates a possible bug elsewhere in the application.  console.error was probably not necessary, though.

Reply:
Oh, I didn't know about console.warn, I'll use that in the future. I'm don't think that we should fire off any warnings here because users can call this code a lot - in the cart screen they can decrement the amount of things in their cart or make it a negative number.

Reply:
Ah, fair enough.  If this is validating user input and preventing them from going negative on quantity then agreed, no warning needed.